nitric/proto/batch/v1/batch.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+package nitric.proto.batch.v1;
+
+import "google/protobuf/struct.proto";
+import "nitric/proto/resources/v1/resources.proto";
+
+// protoc plugin options for code generation
+option go_package = "github.com/nitrictech/nitric/core/pkg/proto/batch/v1;batchpb";
+option java_package = "io.nitric.proto.batch.v1";
+option java_multiple_files = true;
+option java_outer_classname = "BatchService";
+option php_namespace = "Nitric\\Proto\\Batch\\V1";
+option csharp_namespace = "Nitric.Proto.Batch.v1";
+
+// Service for scheduling callbacks on a cadence
+service Batch {
+  rpc CreateJob(CreateJobRequest) returns (CreateJobResponse);
+}
+
+message JobResourceRequirements {
+    // The number of CPUs to allocate for the job
+    int64 cpus = 1;
+    // The amount of memory to allocate for the job
+    int64 memory = 2;
+    // The number of GPUs to allocate for the job
+    int64 gpus = 3;
+}
+
+// ClientMessages are sent from the service to the nitric server
+message CreateJobRequest {
+    // The name of the job to create
+    string name = 1;
```

Comment:
Just a question. Is this repeated in the requirements?

Reply:
That is deploy time, this is runtime. Kind of like how we use the name of a topic to publish to it while also giving it a resource name.